### PR TITLE
Chapter 2: How Programs Work created

### PR DESCRIPTION
**Pull request by speudusa**
There is a lot of new information in this chapter.

### CODE REVIEW

File: content/how-programs-work/_index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Major Concepts 
+
+## Key Terminology
+
+### Introduction
```
Comment (linwang314) [inline content/how-programs-work/_index.md:19]: This link doesn't currently go to the corresponding sub-chapter

File: content/how-programs-work/_index.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Major Concepts 
+
```
Comment (linwang314) [inline content/how-programs-work/_index.md:16]: I'm not sure whether this serves as the content page, but there seems to be a couple problems if it were: 1) under Headings Major Concepts and Key Terminology, there isn't any other content; 2) the subheadings are not consistent with the the structure of each sub-chapter. For example, "Introduction" has "Program" and "Algorithm" here, but in the "Introduction" sub-chapter, there is only "Algorithms" section but not "Program" section.

File: content/how-programs-work/reading/your-first-program/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+   
+   A. The repo will have the same name, but below it will be a link to the original repo.
+
+   B. Click the green _<> Code_ button.  This will open a new dropdown menu.
+
+   C. In the dropdown menu, select _HTTPS_.  A URL will appear. Copy that URL.
+
+
+
+GitHub has documentation on [Forking a repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) and [Cloning your forked repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo) if you would like more guidance.
```
Comment (linwang314) [inline content/how-programs-work/reading/your-first-program/index.md:90]: Add "#cloning-your-forked-repository" to the hyperlink "Cloning your forked repository" to go to the right section.

File: content/how-programs-work/_index.md (hunk 2 above)
Comment (jwoolbright23) [reply content/how-programs-work/_index.md:24]: Fixed both of the above issues.

File: content/how-programs-work/reading/your-first-program/index.md (hunk 3 above)
Comment (jwoolbright23) [reply content/how-programs-work/reading/your-first-program/index.md:90]: Fixed the above issue.

File: content/installations/install-node/_index.md [hunk 4]
```diff
@@ -5,27 +5,32 @@ draft: false
 weight: 5
 ---
 
-## Setting up Node
+We will need `Node.js` to run JavaScript code in our IDE.
 
+## Setting up Node
+<!-- TODO: Link to terminal chapter -->
 Installing Node on your computer also installs the NPM Command Line Interface (CLI), which allows you to run specialized commands in the [terminal](#terminal-chapter).
```
[the remark this answers is not in the document]
Comment (jwoolbright23) [reply content/installations/install-node/_index.md:12]: fixed the above issue. That specific terminal chapter has not yet been created. Once it is added we can go back and add the link. For the time being we have removed the hyperlink.

File: content/how-programs-work/reading/your-first-program/index.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+`Hello, World` will be your first program as well. Welcome to the club!
+
+1. Open your IDE, and open your new repo.  You should see a file tree with the same files from the repo.
+   {{< rawhtml >}}
+      <img src="../images/filetree.png" alt="file tree for hello-world repo" width=60%/>
+   {{< /rawhtml >}}
+
+1. Double click on `hello.js` to open the file.  You should see instructions and `console.log("Hello");`.  You can type in this file now.  Follow the instructions to complete your first program.  
+   1. Enter your code on line 7 to print 'Hello, World!
+   1. Open a terminal inside Visual Studio Code.  If you need a refresher, checkout the [installation guide]({{< relref "../../../installations/terminal-setup/_index.md" >}}).
```
[the remark this answers is not in the document]
Comment (jwoolbright23) [reply content/how-programs-work/reading/your-first-program/index.md:132]: resolved the above issue.

File: content/how-programs-work/reading/your-first-program/index.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+   remote: Resolving deltas: 100% (3/3), completed with 1 local object.
+   To https://github.com/USERNAME/js-hello-world.git
+      408d555..b1bb194  main -> main
+   computer:js-hello-world username$ 
+   ```
+
+   1. Line 1 `git status` used to check on the latest version of the project. 
+   1. Line 6 notes Changes not staged for commit.  We need to add these.
+   1. Line 17 `git add .` will add these to GitHub when we are ready to push up.
+   1. Line 18 `git commit -m "now says Hello World"` A commit is like taking a photo of the latest version of your code.  The message (`-m`) acts like a caption to keep track of the changes.
```
[the remark this answers is not in the document]
Comment (jwoolbright23) [reply content/how-programs-work/reading/your-first-program/index.md:204]: resolved the above issue.

**Review by gildedgardenia — APPROVED** [review]
Added some info on how to fetch an upstream! Please read before merging!

**Pull request merged**